src/Internal/ContextClusterWorker.php

```diff
@@ -109,7 +109,16 @@ public function run(): void
                 };
             }
 
-            $this->joinFuture->await(new TimeoutCancellation(ClusterWatcher::WORKER_TIMEOUT));
+            try {
+                $result = $this->joinFuture->await(new TimeoutCancellation(ClusterWatcher::WORKER_TIMEOUT));
+            } catch (CancelledException) {
+                $this->close();
+                // Give it a second to reap the result. Generally this never should time out, unless something is seriously broken.
+                $result = $this->joinFuture->await(new TimeoutCancellation(1));
+            }
+            if ($result instanceof \Throwable) {
```

Comment:
I wouldn't expect `$result` to be a Throwable. If ExitFailure is returned, the `ContextException` is thrown.

Reply:
Yes, that's a mistake.